store/tikv/kv.go

```diff
@@ -347,3 +395,60 @@ func (s *KVStore) SetTiKVClient(client Client) {
 func (s *KVStore) GetTiKVClient() (client Client) {
 	return s.client
 }
+
+func (s *KVStore) getMinResolveTSByStores(stores []*Store) uint64 {
+	failpoint.Inject("injectResolveTS", func(val failpoint.Value) {
+		injectTS := val.(int)
+		failpoint.Return(uint64(injectTS))
+	})
+	minSafeTS := uint64(math.MaxUint64)
+	s.resolveTSMu.RLock()
+	defer s.resolveTSMu.RUnlock()
+	for _, store := range stores {
+		safeTS := s.resolveTSMu.resolveTS[store.storeID]
+		if safeTS < minSafeTS {
+			minSafeTS = safeTS
+		}
+	}
+	return minSafeTS
+}
+
+func (s *KVStore) safeTSUpdater() {
+	t := time.NewTicker(time.Second * 2)
+	defer t.Stop()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	for {
+		select {
+		case <-s.Closed():
+			return
+		case <-t.C:
+			s.updateResolveTS(ctx)
+		}
+	}
+}
+
+func (s *KVStore) updateResolveTS(ctx context.Context) {
+	stores := s.regionCache.getStoresByType(tikvrpc.TiKV)
```

Comment:
All stores? Why not the stores in the same zone?

Reply:
When `txnScope` is global, we need to startTS with all the stores resolvedTS, so we need to maintian all the resolvedTS for each tikv.